mozilla-central-patches/aboutnewtabservice.patch

```diff
@@ -0,0 +1,226 @@
+diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
+--- a/browser/app/profile/firefox.js
++++ b/browser/app/profile/firefox.js
+@@ -1271,4 +1271,7 @@ pref("browser.newtabpage.activity-stream
+ pref("browser.newtabpage.activity-stream.prerender", true);
+ pref("browser.newtabpage.activity-stream.aboutHome.enabled", true);
++#ifndef RELEASE_OR_BETA
++pref("browser.newtabpage.activity-stream.debug", false);
++#endif
+ 
+ pref("browser.library.activity-stream.enabled", true);
+diff --git a/browser/components/newtab/aboutNewTabService.js b/browser/components/newtab/aboutNewTabService.js
+--- a/browser/components/newtab/aboutNewTabService.js
++++ b/browser/components/newtab/aboutNewTabService.js
+@@ -11,4 +11,6 @@ const {utils: Cu, interfaces: Ci} = Comp
+ Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+ Cu.import("resource://gre/modules/Services.jsm");
++Cu.import("resource://gre/modules/AppConstants.jsm");
++
+ XPCOMUtils.defineLazyModuleGetter(this, "AboutNewTab",
+                                   "resource:///modules/AboutNewTab.jsm");
+@@ -16,4 +18,5 @@ XPCOMUtils.defineLazyModuleGetter(this, 
+ const LOCAL_NEWTAB_URL = "chrome://browser/content/newtab/newTab.xhtml";
+ 
++const BASE_AS_PATH = "resource://activity-stream/data/content/activity-stream";
+ const ACTIVITY_STREAM_URL = "resource://activity-stream/data/content/activity-stream.html";
+ 
+@@ -24,7 +27,10 @@ const ABOUT_URL = "about:newtab";
+ const IS_MAIN_PROCESS = Services.appinfo.processType === Services.appinfo.PROCESS_TYPE_DEFAULT;
+ 
++const IS_RELEASE_OR_BETA = AppConstants.RELEASE_OR_BETA;
++
+ // Pref that tells if activity stream is enabled
+ const PREF_ACTIVITY_STREAM_ENABLED = "browser.newtabpage.activity-stream.enabled";
+ const PREF_ACTIVITY_STREAM_PRERENDER_ENABLED = "browser.newtabpage.activity-stream.prerender";
++const PREF_ACTIVITY_STREAM_DEBUG = "browser.newtabpage.activity-stream.debug";
+ 
+ function AboutNewTabService() {
+@@ -32,4 +38,9 @@ function AboutNewTabService() {
+   Services.prefs.addObserver(PREF_ACTIVITY_STREAM_ENABLED, this);
+   Services.prefs.addObserver(PREF_ACTIVITY_STREAM_PRERENDER_ENABLED, this);
++  this.initialized = true;
++
++  if (!IS_RELEASE_OR_BETA) {
++    Services.prefs.addObserver(PREF_ACTIVITY_STREAM_DEBUG, this);
++  }
+   this.toggleActivityStream();
```

Comment:
Probably move the `this.initialized = true` line (from between `addObserver`s) to next to `toggleActivityStream` with a comment that more initialization happens there